lib/DependencyScan/ModuleDependencyScanner.cpp

```diff
@@ -1531,17 +1531,22 @@ void ModuleDependencyScanner::resolveSwiftOverlayDependenciesForModule(
     const auto commandLine = moduleInfo.getCommandline();
     // If the textual interface was built without C++ interop, do not query
     // the C++ Standard Library Swift overlay for its compilation.
-    //
-    // FIXME: We always declare the 'Darwin' module as formally having been built
-    // without C++Interop, for compatibility with prior versions. Once we are certain
-    // that we are only building against modules built with support of
-    // '-formal-cxx-interoperability-mode', this hard-coded check should be removed.
-    if (moduleID.ModuleName == "Darwin" ||
-        llvm::find(commandLine, "-formal-cxx-interoperability-mode=off") !=
+    if (llvm::find(commandLine, "-formal-cxx-interoperability-mode=off") !=
          commandLine.end())
       lookupCxxStdLibOverlay = false;
   }
 
+  // FIXME: We always declare the 'Darwin' module as formally having been built
+  // without C++Interop, for compatibility with prior versions. Once we are certain
+  // that we are only building against modules built with support of
+  // '-formal-cxx-interoperability-mode', this hard-coded check should be removed.
```

Comment:
What does this actually mean? We need to pass this flag when building the Darwin overlays? Or all of the OS/SDK overlays?

Reply:
Is it needed for the newer Darwin modules then? Those are only produced by pretty new versions of the compiler.

Reply:
Ah, yes, I didn't quite catch the distinction then. You're right, we do not need this at all for the new `_DarwinFoundation` modules. Removed from the PR.